### PR TITLE
fix(gce): delete instance in cases of retries

### PR DESCRIPTION
**Pull request by fruch**
since we are creating the instances with retries, we sometimes
get into case the first attempt failed cause of api quota issues
and on the retry we fail to create since instance already exists

this change is remove the limit to retry delete instance only when it's spot
and it delete any type of instance on case of failures related to spot or
to quota issues.

Fixes: #7261

### Testing

- [ ]

### PR pre-checks (self review)

- [ ] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

**Comment by fruch** [comment]
seems like in the case of spot instances, we destroy the instance after each failure, maybe that would be a safer option ?

File: sdcm/cluster_gce.py [hunk 1]
```diff
@@ -428,6 +428,9 @@ def _create_node_with_retries(self,
         try:
             return create_instance(**create_node_params)
         except google.api_core.exceptions.GoogleAPIError as gbe:
+            if 'already exists' in str(gbe):
+                return self._get_instances_by_name(name=name, dc_idx=dc_idx)
```
Comment (vponomaryov) [inline sdcm/cluster_gce.py:432]: As I understand, it is useful on the 2nd+ attempt.
If so, then shouldn't we call that `_get_instances_by_name` method first and only after nothing found call the `create_instance` function?

Comment (soyacz) [reply sdcm/cluster_gce.py:432]: I think this happens when retrying instance creation and found that instance was already created

**Review by soyacz — DISMISSED** [review]
LGTM

**Review by soyacz — APPROVED** [review]
LGTM

**Review by vponomaryov — CHANGES_REQUESTED** [review]
Need to update commit message because it is misleading after the latest PR update.

The change itself LGTM

**Comment by fruch** [comment]
> Need to update commit message because it is misleading after the latest PR update.
> 
> The change itself LGTM

update the commit message and the PR description

**Pull request merged**